packages/bridge/contracts/l2/L2DaiGateway.sol

```diff
@@ -78,49 +96,94 @@ contract L2DaiGateway is L2ArbitrumGateway {
     emit Closed();
   }
 
-  function handleNoContract(
-    address l1ERC20,
-    address expectedL2Address,
-    address _from,
-    address _to,
-    uint256 _amount,
-    bytes memory gatewayData
-  ) internal virtual override returns (bool shouldHalt) {
-    // it is assumed that the custom token is deployed in the L2 before deposits are made
-    // trigger withdrawal
-    createOutboundTx(_from, _amount, gatewayData);
-    return true;
+  function outboundTransfer(
+    address l1Token,
+    address to,
+    uint256 amount,
+    bytes calldata data
+  ) public virtual returns (bytes memory) {
+    return outboundTransfer(l1Token, to, amount, 0, 0, data);
   }
 
-  function calculateL2TokenAddress(address l1ERC20) public view virtual override returns (address) {
-    require(l1ERC20 == l1Dai, "L2DaiGateway/token-not-dai");
-    return l2Dai;
+  function outboundTransfer(
+    address l1Token,
+    address to,
+    uint256 amount,
+    uint256, // maxGas
+    uint256, // gasPriceBid
+    bytes calldata data
+  ) public returns (bytes memory res) {
+    require(isOpen == 1, "L2DaiGateway/closed");
+    require(l1Token == l1Dai, "L2DaiGateway/token-not-dai");
+
+    (address from, bytes memory extraData) = parseOutboundData(data);
+    require(extraData.length == 0, "L2DaiGateway/call-hook-data-not-allowed");
+
+    // exit number used for tradeable exits
+    uint256 currExitNum = exitNum;
+
+    Mintable(l2Dai).burn(from, amount);
+
+    // we override the res field to save on the stack
+    res = getOutboundCalldata(l1Token, from, to, amount, extraData);
+    exitNum++;
```

Comment:
I would suggest incrementing the `exitNum` inside of the `getOutboundCalldata` function.